File: pset6/mario/less/mario.py
import cs50

# asks user for the height of the pyramid.
pyramidheight = cs50.get_int("Height: ")
# asks user for the height again if height is not an integer between 1 and 8 inclusive.
while pyramidheight < 1 or pyramidheight > 8:
    pyramidheight = cs50.get_int("Height: ")

# prints pyramid based on height.
for i in range(0, pyramidheight):
    # prints spaces while printing a row of pyramid.
    for s in range(pyramidheight-1, i, -1):
        print(" ", end="")
    # prints hashes while printing a row of pyramid.
    for h in range(0, i + 1):
        print("#", end="")
    # prints next line at the end of a row of pyramid.
    print()

File: pset1/mario/more/mario.c
#include<cs50.h>
#include<stdio.h>

void printpyramid(int pyramidheight);

int main(void)
{
    int height;

    //asks user for height between 1 to 8 and asks again if height is less and 1 or greater than 8.
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    //prints the pyramid based on height that the user has typed in.
    printpyramid(height);
}

void printpyramid(int pyramidheight)
{
    //prints a row of the pyramid based on height.
    for (int i = 0; i < pyramidheight; i = i + 1)
    {
        /*for(int j=0;j<=i;j=j+1){
            printf("#");
        }*/

        //adds a space([-]) while printing a row.
        for (int s = pyramidheight - 1; s > i; s = s - 1)
        {
            //printf(".");
            printf(" ");
        }

        //adds a hash(#) for the left side pyramid while printing a row.
        for (int l = 0; l <= i; l = l + 1)
        {
            printf("#");
        }

        //printf("..");

        //adds two spaces("[-][-]") while printing a row.
        printf("  ");

        //adds a hash(#) for the right side pyramid while printing a row.
        for (int r = 0; r <= i; r = r + 1)
        {
            printf("#");
        }

        //adds a new line("\n") to end a row.
        printf("\n");
    }
}